frontend/src/lib/utils.tsx

```diff
@@ -371,16 +372,71 @@ export const genericOperatorMap: Record<string, string> = {
     is_not_set: '✕ is not set',
 }
 
+export const stringOperatorMap: Record<string, string> = {
+    exact: '= equals',
+    is_not: "≠ doesn't equal",
+    icontains: '∋ contains',
+    not_icontains: "∌ doesn't contain",
+    regex: '∼ matches regex',
+    not_regex: "≁ doesn't match regex",
+    is_set: '✓ is set',
+    is_not_set: '✕ is not set',
+}
+
+export const numericOperatorMap: Record<string, string> = {
+    exact: '= equals',
+    is_not: "≠ doesn't equal",
+    regex: '∼ matches regex',
+    not_regex: "≁ doesn't match regex",
+    gt: '> greater than',
+    lt: '< lower than',
+    is_set: '✓ is set',
+    is_not_set: '✕ is not set',
+}
+
 export const dateTimeOperatorMap: Record<string, string> = {
     is_date_before: '< before',
     is_date_after: '> after',
     is_set: '✓ is set',
     is_not_set: '✕ is not set',
 }
 
+export const booleanOperatorMap: Record<string, string> = {
+    exact: '= equals',
+    is_not: "≠ doesn't equal",
+    is_set: '✓ is set',
+    is_not_set: '✕ is not set',
+}
+
 export const allOperatorsMapping: Record<string, string> = {
     ...dateTimeOperatorMap,
+    ...stringOperatorMap,
+    ...numericOperatorMap,
     ...genericOperatorMap,
+    ...booleanOperatorMap,
+    // slight overkill to spread all of these into the map
+    // but gives freedom for them to diverge more over time
+}
+
+const operatorMappingChoice: Record<keyof typeof PropertyType, Record<string, string>> = {
+    DateTime: dateTimeOperatorMap,
+    String: stringOperatorMap,
+    Numeric: numericOperatorMap,
+    Boolean: booleanOperatorMap,
+}
+
+export function chooseOperatorMap(
+    propertyType: PropertyType | undefined,
+    allowQueryingEventsByDateTime: boolean
+): Record<string, string> {
+    let choice = genericOperatorMap
+    if (propertyType) {
+        choice = operatorMappingChoice[propertyType] || genericOperatorMap
+        if (choice == dateTimeOperatorMap && !allowQueryingEventsByDateTime) {
```

Comment:
nit `choice === dateTimeOperatorMap`

Reply:
> why not run ... directly inside OperatorValueSelect

I was getting annoyed with loading the application and finding specific properties to test it was working

If I lifted the logic out then I could put it as a static component in storybook

This is probably very lazy of me :)

Reply:
> as I guess you know some users have hundreds of thousands properties and we can't really rely on them being loaded into the model. However, since redoing that is outside the scope here, fine to keep this.

I think this definitely needs fixing

I was wanting to make time to check if selectors can return promises... Am imagining the logic returns an immediately returning promise to a property definition if it is in the cache. Or returns a promise, makes an API call, caches the result, and then resolves the promise. The caller doesn't have to make a different decision based on whether it htinks the property exists...

I think we already do that (or similar) for paging the properties

(I have not spent much time sense-checking this idea :))